File: src/main/kotlin/at/fhhgb/mc/component/sim/positioning/IPositioning.java
package at.fhhgb.mc.component.sim.positioning;

import at.fhhgb.mc.component.sim.model.Position;
import at.fhhgb.mc.component.sim.model.recorder.network.AccessPoint;

import java.util.List;

public interface IPositioning {
    public Position calculatePosition(List<AccessPoint> accessPointList);
}

File: src/main/kotlin/at/fhhgb/mc/component/sim/positioning/IPositioning.kt
package at.fhhgb.mc.component.sim.positioning

import at.fhhgb.mc.component.sim.model.Position
import at.fhhgb.mc.component.sim.model.recorder.network.AccessPoint

/**
 * An interface for calculating positions.
 */
interface IPositioning {
    /**
     * Calculates a sector-based position using detected access points.
     *
     * @param accessPointList a list of detected access points.
     * @return a sector-based position
     */
    fun calculatePosition(accessPointList: List<AccessPoint>): Position
}
